pkg/controller/core.oam.dev/v1alpha2/applicationconfiguration/render.go

```diff
@@ -342,6 +322,30 @@ func isRevisionEnabled(traitDefs []v1alpha2.TraitDefinition) bool {
 	return false
 }
 
+// setAppWorkloadInstanceName sets the name of the workload instance depends on the component revision
+// and the workload kind
+func setAppWorkloadInstanceName(componentName string, w *unstructured.Unstructured, revision int) {
+	// TODO: we can get the workloadDefinition name from w.GetLabels()["oam.WorkloadTypeLabel"]
+	// and use a special field like "use-inplace-upgrade" in the definition to allow configurable behavior
+
+	// we hard code the behavior depends on the workload group/kind for now. The only in-place upgradable resources
+	// we support is cloneset for now. We can easily add more later.
+	if w.GroupVersionKind().Group == kruise.GroupVersion.Group {
+		if w.GetKind() == reflect.TypeOf(kruise.CloneSet{}).Name() {
```

Comment:
also consider statefulset

Reply:
I was going to but not 100% sure if statefulset should be upgraded in-place